streams-tests/shared/src/test/scala/zio/stream/ZStreamSpec.scala

```diff
@@ -208,6 +209,49 @@ object ZStreamSpec extends ZIOBaseSpec {
             assertM(ZStream(1, 2, 3).aggregate(t).runCollect.either)(isLeft(equalTo(fail)))
           }
         ),
+        suite("debounce")(
+          testM("drop earlier elements within waitTime") {
+            assertWithChunkCoordination(List(Chunk(1, 2), Chunk(3, 4), Chunk.single(5))) { c =>
+              val stream = ZStream
+                .fromQueue(c.queue)
+                .collectWhileSuccess
```

Comment:
We can extract this bit out into the stepper - have it take a function `ZStream[A] => ZStream[B]` that is applied to `ZStream.fromQueue` and then fed into `.tap(verifyAssertions)`.